ggml-metal.m

```diff
@@ -660,7 +662,11 @@ void ggml_metal_graph_compute(
 
                                             nth0 = 8;
                                             nth1 = 8;
-                                            [encoder setComputePipelineState:ctx->pipeline_mul_mat_q4_0_f32];
+                                            if (ne01 % 8 == 0) {
```

Comment:
Is it somehow possible to remove this restriction without affecting the performance?
If so, we can simply delete the old `pipeline_mul_mat_q4_0_f32` kernel

Reply:
This is for models like WizardLM. In these models the last mat-vec multiplication in inference will have row number = 32001, while our new kernel consumes 8 rows every time. Let me test later if adding capabilities for dealing with <8 rows in the kernel will make it run slower or faster.

Reply:
Removed the old kernel. Might have minor performance gain or loss depending on model size and n_vocab. Tested on llama-7B and wizardlm-30B, generation results are same.